src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -203,6 +206,32 @@ public Task<RestoreResult> ExecuteAsync()
                 restoreTime.Elapsed);
         }
 
+        private void ReplayWarningsAndErrors()
+        {
+            var logMessages = _request?.ExistingLockFile?.LogMessages;
+
+            if (logMessages != null)
+            {
+                foreach (var logMessage in logMessages)
+                {
+                    var restoreLogMessage = new RestoreLogMessage(logMessage.Level, logMessage.Code, logMessage.Message)
```

Comment:
Why do you need to do this?
Why can't you just call 
_request.Log.Log(logMessage) ?